app/models/concerns/audit_extensions.rb

```diff
@@ -65,8 +65,8 @@ def fix_auditable_type
     # STI Host class should use the stub module instead of Host::Base
     self.auditable_type = "Host"          if auditable_type =~  /Host::/
     self.associated_type = "Host"         if associated_type =~ /Host::/
-    self.auditable_type = auditable.type  if auditable_type == "Taxonomy" && auditable
-    self.associated_type = auditable.type if auditable_type == "Taxonomy" && auditable
+    self.auditable_type = auditable.type if auditable && ["Taxonomy", "LookupKey"].include?(auditable_type)
+    self.associated_type = auditable.type if auditable && ["Taxonomy", "LookupKey"].include?(auditable_type)
```

Comment:
after some debugging, looks like while this condition is always false, actually a side effect of the `auditable` call is what caused this to work correctly, possibly because it led to a reload of the auditable object. Otherwise the save fails. Some preliminary testing indicated that moving this callback from `after_validation` to `before_save` allows removing the second `auditable` call and saving the object correctly, but this needs to be checked to make sure this doesn't cause other things to break.